pkg/testing/integration/program.go

```diff
@@ -153,6 +153,9 @@ type ProgramTestOptions struct {
 	ExpectRefreshChanges bool
 	// SkipRefresh indicates that the refresh step should be skipped entirely.
 	SkipRefresh bool
+	// SkipStackRemoval indicates that the stack should not be removed. (And so the test's results could be inspected
+	// in the Pulumi Service after the test has completed.)
+	SkipStackRemoval bool
```

Comment:
Never for tests that are checked in, since obviously it would just leak resources on the Pulumi Service. However, I found it useful when debugging this particular PR, so I figured I'd keep the code in for later use.

(I wasn't sure if I could actually trust the performance numbers I was seeing, and wanted to confirm that the engine events were actually persisted on the service. Which requires us skipping the cleanup step at the end.)